internal/repositories/dcn.go

```diff
@@ -14,14 +17,44 @@ func DCNToAPI(d *models.DCN) *gmodel.Dcn {
 		Node:      d.Node,
 		ExpiresAt: d.Expiration.Ptr(),
 		Name:      d.Name.Ptr(),
+		VehicleID: d.VehicleID.Ptr(),
 	}
 }
 
-func (r *Repository) GetDCNByNode(ctx context.Context, node []byte) (*gmodel.Dcn, error) {
-	dcn, err := models.DCNS(models.DCNWhere.Node.EQ(node)).One(ctx, r.pdb.DBS().Reader)
+func (r *Repository) getDCNByNode(ctx context.Context, dcnParams model.DCNBy) (*gmodel.Dcn, error) {
+	dcn, err := models.DCNS(
+		models.DCNWhere.Node.EQ(dcnParams.Node),
+	).One(ctx, r.pdb.DBS().Reader)
 	if err != nil {
 		return nil, err
 	}
 
 	return DCNToAPI(dcn), nil
 }
+
+func (r *Repository) getDCNByName(ctx context.Context, dcnParams model.DCNBy) (*gmodel.Dcn, error) {
+	dcn, err := models.DCNS(
+		models.DCNWhere.Name.EQ(null.StringFrom(*dcnParams.Name)),
+	).One(ctx, r.pdb.DBS().Reader)
+	if err != nil {
+		return nil, err
+	}
+
+	return DCNToAPI(dcn), nil
+}
+
+func (r *Repository) GetDCN(ctx context.Context, params model.DCNBy) (*gmodel.Dcn, error) {
+	if params.Name != nil && len(params.Node) > 0 {
+		return nil, fmt.Errorf("please provide one of Name or Node but not both")
```

Comment:
This is terribly picky but

1. I wouldn't include "please" in a repository error
2. We don't need `fmt` because we're not interpolating anything